### DISCUSSION TITLE
Does useBalance only retrieve eth mainnet balance?

### DISCUSSION THREAD

**Discussion by lsbyerley** [wevm/wagmi#87] [Q&A] [answered]
It appears the `useBalance` hook only returns the balance of eth mainnet, even when switching to other networks such Eth Rinkeby or Polygon Mainnet. Is this a bug or are there plans to update this hook to retrieve the balance of the active network?

**Comment by tmm** [comment, marked as answer]
It should work for all networks. Does your [ethers.js Provider](https://wagmi-xyz.vercel.app/docs/provider#provider-optional) use the current chain id?

```tsx
import { providers } from 'ethers'
import { Provider } from 'wagmi'

// Use `chainId` to make ethers.js provider network-aware
const provider = ({ chainId }) =>
  new providers.InfuraProvider(chainId, 'Your infura id')

const App = () => (
  <Provider provider={provider}>
    <YourRoutes />
  </Provider>
)
```

**Reply by lsbyerley** [reply]
Ahh i see. I had not setup the optional provider so it was using the default. Thanks for the reply. The balances are updating now!

**Comment by lsbyerley** [comment]
I still can't seem to get it to pull polygon mainnet balance though. I'm using this provider

```
const provider = ({ chainId }) =>
    new providers.AlchemyProvider(chainId, alchemyId);
```

**Reply by lsbyerley** [reply]
Actually.. i think it is pulling it in i just didn't have a balance lol.. i was thrown off because it still said ETH instead of MATIC

**Reply by tmm** [reply]
Good point. Could be cool to update symbol based on network ([hard-code right now](https://github.com/tmm/wagmi/blob/main/packages/react/src/hooks/accounts/useBalance.ts#L79)).

**Reply by lsbyerley** [reply]
Is the token symbol something that can be pulled from ether's (or another library) or does it need to be a hardcoded value map?

**Reply by tmm** [reply]
Pushed an update that uses the active chain's [native currency](https://github.com/tmm/wagmi/blob/main/packages/private/src/constants/chains.ts) property.

If you [pass in an ERC-20 token address](https://wagmi-xyz.vercel.app/docs/hooks/useBalance#token-optional) to `useBalance`, the symbol is fetched from the contract.

**Reply by lsbyerley** [reply]
Nice... i _think_ this may have broken something when using this library with nextjs.. still trying to pinpoint the issue. Will report back if it is the cause

**Reply by lsbyerley** [reply]
I downgraded to `0.1.3` and am still seeing an error so must be something on my end. Thanks for the quick update on the symbol!

**Reply by tmm** [reply]
Haven't published the new version yet. Will do shortly.

**Reply by lsbyerley** [reply]
Thank you! I'm assuming we should be using the `networkData.chain.nativeCurrency.symbol` field from useNetwork now instead of the `balanceData.symbol` property from useBalance when showing the currency symbol?

**Reply by tmm** [reply]
`balanceData.symbol` should return the correct symbol. (You could use `networkData` instead, but would be more verbose in this case.)

**Reply by JanRuettinger** [reply]
I just tried it with another chain. I get the correct amount but the symbol is still ETH when I use balanceData.symbol.

This is how I define the chain:
```
const chains = [
  {
    id: 1666700000,
    name: 'Harmony Testnet Shard 0',
    rpcUrls: ['https://api.s0.b.hmny.io'],
    nativeCurrency: { name: 'ONE', symbol: 'ONE', decimals: 18 as 18 },
    blockExplorers: [
      {
        name: 'Harmony Block Explorer',
        url: 'https://explorer.harmony.one',
        standard: 'EIP3091',
      },
    ],
    testnet: false,
  },
];
```
